Source/Core/Duality/Utility/Coroutines/Coroutine.cs

```diff
@@ -0,0 +1,145 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Duality.Resources;
+
+namespace Duality.Utility.Coroutines
+{
+	/// <summary>
+	/// A Coroutine's management class
+	/// </summary>
+	public sealed class Coroutine : IDisposable
+	{
+		/// <summary>
+		/// Starts a coroutine and registers it in the scene's CoroutineManager
+		/// </summary>
+		/// <param name="scene">The scene that will execute the coroutine</param>
+		/// <param name="method">The coroutine implementation</param>
+		/// <param name="name">The name of the coroutine, optional</param>
+		/// <returns>The created corutine</returns>
+		public static Coroutine Start(Scene scene, IEnumerable<WaitUntil> method, string name = null)
+		{
+			return Start(scene, method.GetEnumerator(), name);
+		}
+
+		/// <summary>
+		/// Starts a coroutine and registers it in the scene's CoroutineManager
+		/// </summary>
+		/// <param name="scene">The scene that will execute the coroutine</param>
+		/// <param name="enumerator">The coroutine IWaitConditions enumerator</param>
+		/// <param name="name">The name of the coroutine, optional</param>
+		/// <returns>The created corutine</returns>
+		public static Coroutine Start(Scene scene, IEnumerator<WaitUntil> enumerator, string name = null)
+		{
+			return scene.CoroutineManager.StartNew(enumerator, name);
+		}
+
+		private IEnumerator<WaitUntil> enumerator = null;
+		private WaitUntil currentCondition;
+
+		/// <summary>
+		/// The Coroutine's current <see cref="CoroutineStatus">execution status</see>
+		/// </summary>
+		public CoroutineStatus Status { get; private set; }
+
+		/// <summary>
+		/// The Exception encountered during the last frame
+		/// </summary>
+		public Exception LastException { get; private set; }
+
+		/// <summary>
+		/// The Coroutine's Name
+		/// </summary>
+		public string Name { get; private set; }
+
+		/// <summary>
+		/// A Tag object
+		/// </summary>
+		public object Tag { get; private set; }
```

Comment:
Is never used?

Reply:
Just a Tag object.. might come in handy for someone?
Ah, yeah.. should have a public setter as well

Reply:
If we don't currently have a good usecase I think its better to just remove it to avoid confusion about the API